replicatedhq/ship
# handle super long errors with fallback "check logs" message

What I Did
------------
Check the length of an error message and if it's great than 500 characters, display a default "Check the logs" message. The reason for this is that sometimes there are insanely long errors that should be suppressed in the UI but still give the user a pathway to view the full error if they want.

How I Did it
------------
Check the length of a steps detail message and display the right message depending on it's length

How to verify it
------------
To view the actual error message run through the ship init flow after you already have a `base` directory. On the 3rd step when ship is building your assets you should receive an error and it should be displayed in the browser. If you look in the network tab at the most recent `render` call the message should match what is in `progress.detail`


To view the overwritten error message, run the following command:
```
docker run -p 8800:8800 --rm -it -v `pwd`:/out \
  -v /var/run/docker.sock:/var/run/docker.sock \
  -e HTTP_PROXY -e HTTPS_PROXY -e NO_PROXY \
  replicated/ship:alpha app \
  --customer-id=mxU8plZ0gL2Ieh9G4k2RYUne_DI-AYR2 \
  --installation-id=r9jNofHya8KltH4OVq4GcY0ab6fsaYqk
``` 
On the 3rd step when ship is building your assets you should receive an error that says
"There was an unexpected error! Please check .ship/debug.log for more details".
If you look in the network tab at the most recent `render` call the actual error message in in `progress.detail` is a giant string that we are overriding.


Description for the Changelog
------------
Handle long error message better
Add icons to success and error message screens.


Picture of a Boat (not required but encouraged)
------------
🚢
